borg/upgrader.py

```diff
@@ -267,10 +267,14 @@ def find_key_file(cls, repository):
         get_keys_dir = cls.get_keys_dir
         id = hexlify(repository.id).decode('ascii')
         keys_dir = get_keys_dir()
+        def ex():
+            raise KeyfileNotFoundError(repository.path, keys_dir)
+        if not os.path.exists(keys_dir):
+            ex()
         for name in os.listdir(keys_dir):
             filename = os.path.join(keys_dir, name)
             with open(filename, 'r') as fd:
                 line = fd.readline().strip()
                 if line and line.startswith(cls.FILE_ID) and line[10:] == id:
                     return filename
-        raise KeyfileNotFoundError(repository.path, get_keys_dir())
+        ex()
```

Comment:
maybe do a amend commit to fix above.

also mention "....., fixes #576" in the commit comment, so it automatically closes the issue.